### PR TITLE
Merged /entity/overview and /entity/details into one endpoint /state/…

### PR DESCRIPTION
**Pull request by KrzysztofLabus-RDX**
…entity/details

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.GatewayApi/Validators/EntityDetailsRequestValidator.cs [hunk 1]
```diff
@@ -67,15 +67,24 @@
...
         RuleFor(x => x.AtLedgerState)
             .SetValidator(ledgerStateSelectorValidator);
+
+        RuleFor(x => x.Addresses)
+            .NotEmpty()
+            .DependentRules(() =>
+            {
+                RuleFor(x => x.Addresses.Count)
+                    .GreaterThan(0)
+                    .LessThan(5);
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.GatewayApi/Validators/EntityDetailsRequestValidator.cs:83]: thought we wanted to support only 1 address for now?

File: src/RadixDlt.NetworkGateway.GatewayApi/Handlers/DefaultEntityHandler.cs [hunk 2]
```diff
@@ -92,18 +93,28 @@ public DefaultEntityHandler(ILedgerStateQuerier ledgerStateQuerier, IEntityState
...
-    public async Task<GatewayModel.EntityDetailsResponse?> Details(GatewayModel.EntityDetailsRequest request, CancellationToken token = default)
+    public async Task<GatewayModel.StateEntityDetailsResponse> Details(GatewayModel.StateEntityDetailsRequest request, CancellationToken token = default)
     {
         var ledgerState = await _ledgerStateQuerier.GetValidLedgerStateForReadRequest(request.AtLedgerState, token);
 
-        return await _entityStateQuerier.EntityDetailsSnapshot((GlobalAddress)request.Address, ledgerState, token);
-    }
+        var items = await Task.WhenAll(request.Addresses.Select(async address =>
+        {
+            // TODO exception driven logic flow is a very bad design
```
[the remark this answers is not in the document]
Comment (KrzysztofLabus-RDX) [reply src/RadixDlt.NetworkGateway.GatewayApi/Handlers/DefaultEntityHandler.cs:102]: Rolled back.

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 3]
```diff
@@ -304,54 +304,30 @@ paths:
...
-        "4XX":
-          $ref: "#/components/responses/ErrorResponse"
-  /entity/details:
-    post:
-      operationId: EntityDetails
+      operationId: StateEntityDetails
       summary: Get Entity Details
       description: |
-        Returns detailed information for a single entity.
+        Returns detailed information for a each entity.
```
[the remark this answers is not in the document]
Comment (KrzysztofLabus-RDX) [reply src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml]: Fixed

File: src/RadixDlt.NetworkGateway.GatewayApi/Validators/EntityDetailsRequestValidator.cs (hunk 1 above)
Comment (KrzysztofLabus-RDX) [reply src/RadixDlt.NetworkGateway.GatewayApi/Validators/EntityDetailsRequestValidator.cs:83]: Reconfigured for exactly 1.

**Pull request merged**